common/network-common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -79,14 +80,27 @@ public static String bytesToString(ByteBuffer b) {
     return Unpooled.wrappedBuffer(b).toString(StandardCharsets.UTF_8);
   }
 
-  /*
+  /**
    * Delete a file or directory and its contents recursively.
    * Don't follow directories if they are symlinks.
-   * Throws an exception if deletion is unsuccessful.
+   *
+   * @param file Input file / dir to be deleted
+   * @throws IOException if deletion is unsuccessful
    */
   public static void deleteRecursively(File file) throws IOException {
     if (file == null) { return; }
 
+    // On Unix systems, use operating system command to run faster
+    // If that does not work out, fallback to the Java native way
+    if (SystemUtils.IS_OS_UNIX) {
+      try {
+        deleteRecursivelyForUnix(file);
+        return;
+      } catch (IOException e) {
+        // ignore and fall back to the Java native way
+      }
+    }
+
     if (file.isDirectory() && !isSymlink(file)) {
```

Comment:
can we move this also into a private function named deleteRecursivelyUsingJavaIO